packages/core/src/main/Account.ts

```diff
@@ -265,6 +254,10 @@ export class Account extends EventEmitter {
       const localClient = await this.loadAndValidateLocalClient();
       return {isNewClient: false, localClient};
     } catch (error) {
+      if (!clientInfo) {
+        // If no client info provided, the client should not be created
+        throw error;
```

Comment:
This is the main change (all the rest is just documentation and moving methods around)